crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -505,40 +521,22 @@ impl RecursorDnsHandle {
                 continue;
             }
 
-            let cached_a = self.record_cache.get(
```

Comment:
Would be nice to split this into multiple commits:

- Rewriting this logic to be less repetitive
- Splitting it out into a function
- Using the function in new places

Reply:
I added an additional intermediate commit that introduces the HashMap before extracting the helper method